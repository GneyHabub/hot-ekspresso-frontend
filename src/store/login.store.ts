import { makeAutoObservable, runInAction } from "mobx";
import { login } from "../api/authentication";
import { FetchingStatus } from "../utils/types";
import { notificationStore } from "./notification.store";
import store from "./store";

class LoginStore {
  fetchingStatus: FetchingStatus = "notFetched";
  email: string = '';
  emailDirty: boolean = false;
  password: string = '';
  passwordDirty: boolean = false;
  constructor() {
    makeAutoObservable(this);
    this.submit = this.submit.bind(this);
  }

  setEmail(email: string) {
    this.email = email;
  }
  dirtyEmail() {
    this.emailDirty = true;
  }
  setPassword(password: string) {
    this.password = password;
  }
  dirtyPassword() {
    this.passwordDirty = true;
  }
  async submit() {
    runInAction(() => {
      this.fetchingStatus = "fetching";
    });
    try {
      const token = await login({email: this.email, password: this.password});
      store.setToken(token);
      runInAction(() => {
        this.fetchingStatus = "fetching";
      })
    } catch(e) {
      notificationStore.setNotificationFromError(e);
      this.fetchingStatus = "errorFetching";
    }
  }

  get isFetching() {
    return this.fetchingStatus === "fetching"
  }
  get emailError() {
    return !this.email;
  }
  get passwordError() {
    return !this.password;
  }
}

export default new LoginStore();