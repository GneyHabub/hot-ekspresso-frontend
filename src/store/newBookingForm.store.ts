import { makeAutoObservable } from "mobx";

class newBookingFormStore {
  constructor() {
    makeAutoObservable(this);
  }

  airportOptions = [];
  fromAirport: string | undefined = undefined;
  toAirport: string | undefined = undefined;
}

export default new newBookingFormStore();