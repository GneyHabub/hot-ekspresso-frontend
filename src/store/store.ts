import { makeAutoObservable, runInAction } from "mobx";

class AppStore {
  token: string | null = null;
  
  constructor() {
    makeAutoObservable(this);
    this.setToken = this.setToken.bind(this);

    runInAction(() => {
      this.token = localStorage.getItem('token');
    });
  }

  setToken(token: string) {
    localStorage.setItem('token', token);
    this.token = token;
  }
}

export default new AppStore();