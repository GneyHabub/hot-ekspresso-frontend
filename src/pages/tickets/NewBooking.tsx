import { Search } from '@mui/icons-material';
import { MobileDatePicker } from '@mui/lab';
import { Autocomplete, Box, Button, Card, CardContent, CircularProgress, colors, FormControl, IconButton, MenuItem, Select, Stack, TextField, Typography } from '@mui/material';
import { observer } from 'mobx-react';
import React from 'react';
import store from '../../store/newBookingForm.store';
import getEnv from '../../utils/getEnv';
import { useAuthToken } from '../../utils/hooks/useAuthToken';
import { ServiceClasses } from '../../utils/types';
import BookingItem from './BookingItem';
import ArrowForwardIcon from '@mui/icons-material/ArrowForward';
import CancelIcon from '@mui/icons-material/Cancel';
import dayjs from 'dayjs';
import PhoneNumberInput from "material-ui-phone-number";
import { useNavigate } from 'react-router';

const NewBooking: React.FC = observer(() => {
  const authToken = useAuthToken({
    domain: getEnv("DOMAIN"),
    audience: "https://dev-od7-l947.us.auth0.com/api/v2/",
    clientId: getEnv("CLIENT_ID"),
    redirect_uri: window.location.origin,
  });
  const navigate = useNavigate();

  React.useEffect(() => {
    if(authToken) {
      store.fetchAirports(authToken);
    }
    return(() => {
      store.resetForm();
    })
  }, [authToken]);

  React.useEffect(() => {
    if(store.submittingStatus === "fetched") {
      navigate("/tickets");
    }
  }, [store.submittingStatus]);

  if(store.airportsfetchingStatus !== "fetched") {
    return <CircularProgress />
  }

  return (
    <Box
      sx={{
        display: 'flex',
        flexDirection: 'column',
        gap: "15px"
      }}
    >
      <Card>
        <CardContent> 
        <Typography variant="h4" gutterBottom component="div">
          New booking
        </Typography>
        <Box
          sx={{
            display: 'flex',
            flexDirection: 'column',
            rowGap: "20px",
            alignItems: "center",
            "& > .form-block": {
              width: "80%"
            }
          }}
        >
          <FormControl
            sx={{
              display: 'flex',
              flexDirection: 'column',
              rowGap: "15px",
              "& > *": {
                display: 'flex',
                justifyContent: "space-evenly",
                gap: "7px"
              }
            }}
            className="form-block"
          >
            <Box>
              <Autocomplete
                freeSolo
                sx={{
                  width: "50%"
                }}
                getOptionLabel={(airport) => `${airport.iata} - ${airport.city}`}
                options={store.airports}
                renderInput={(params) => <TextField required {...params} label="From" />}
                onChange={(e, value) => store.setFromAirport(value)}
              />
              <Autocomplete
                freeSolo
                sx={{
                  width: "50%"
                }}
                getOptionLabel={(airport) => `${airport.iata} - ${airport.city}`}
                options={store.airports}
                renderInput={(params) => <TextField required {...params} label="To" />}
                onChange={(e, value) => store.setToAirport(value)}
              />
            </Box>
            <Box>
              <MobileDatePicker
                label="Date mobile"
                inputFormat="DD/MM/YYYY"
                value={store.departureDate}
                renderInput={(params) => <TextField {...params} sx={{width: "50%"}} label="Departure Date" />}
                onChange={(value) => store.setDepartureDate(value || new Date())}
              />
              <Select
                sx={{
                  width: "50%"
                }}
                value={store.serviceClass}
                onChange={(e) => store.setServiceClass(e.target.value as ServiceClasses)}
              >
                {["Economy", "Comfort", "Business"].map(serviceClass => <MenuItem key={serviceClass} value={serviceClass}>{serviceClass}</MenuItem>)}
              </Select>
            </Box>
          </FormControl>
          <Button
            variant="contained"
            type="submit"
            disabled={!store.formIsValid}
            startIcon={<Search />}
            onClick={() => {
              store.setSelectedFlight(undefined)
              store.searchFlights(authToken)
            }}
          >
            Search flights
          </Button>
        </Box>
        </CardContent>
      </Card>
      {store.flightsFetchingStatus === "fetching" &&
        <CircularProgress />
      }
      {store.flightsFetchingStatus === "fetched" && store.flights.length !== 0 && !store.selectedFlight &&
        <Card>
          <CardContent> 
            {store.flights.map((b, idx) => (
                <BookingItem idx={idx} booking={b} key={b.id} onPick={(booking) => store.setSelectedFlight(booking)}/>
              ))}
          </CardContent>
        </Card>
      }
      {store.selectedFlight &&
        <Card
          raised
          sx={{
            color: "white",
            background: `linear-gradient(145deg, ${colors.blue[500]}, ${colors.blue.A700})`,
            padding: "20px"
          }}
        >
          <Stack
            direction="row"
            justifyContent="space-between"
          >
            <Stack
              direction="row"
              alignItems="center"
              gap="10px"
            >
              <Typography variant="h4">
                {store.selectedFlight.flights[0].from}
              </Typography>
              <ArrowForwardIcon />
              {store.selectedFlight.flights.slice(1).map(flight => (
                <>
                  <Typography variant="h6">
                    {flight.from}
                  </Typography>
                  <ArrowForwardIcon />
                </>
              ))}
              <Typography variant="h4">
                {store.selectedFlight.flights[store.selectedFlight.flights.length - 1].to}
              </Typography>
            </Stack>

            <IconButton
              onClick={() => store.setSelectedFlight(undefined)}
              sx={{
                color: "white"
              }}
            >
              <CancelIcon fontSize="large" />
            </IconButton>
          </Stack>
          <Typography variant="subtitle1">
            {dayjs(store.selectedFlight.flights[0].departure).format("DD-MM-YYYY")}
          </Typography>
        </Card>
      }
      {store.selectedFlight !== undefined &&
        <>
          {store.passengers.map((p, idx) => (
            <Card key={idx}>
              <CardContent>
                <Box
                  sx={{
                    display: "flex",
                    justifyContent: "space-between"
                  }}
                >
                  <Typography variant="h6" gutterBottom>
                    {`Passenger #${idx + 1}`}
                  </Typography>
                  {idx !== 0 &&
                    <IconButton
                      onClick={() => store.deletePassenger(idx)}
                    >
                      <CancelIcon fontSize="medium" />
                    </IconButton>
                  }
                </Box>
                <Box
                  sx={{
                    display: "grid",
                    gridTemplateColumns: "1fr 1fr",
                    gridTemplateRows: "1fr 1fr",
                    gap: "12px",
                    "& .MuiFilledInput-input": {
                      backgroundColor: "white"
                    },
                    "& .phone-input": {
                      backgroundColor: "white",
                      border: "1px solid rgba(0, 0, 0, 0.23)",
                      borderRadius: "4px"
                    }
                  }}
                >
                  <TextField value={p.name} label="Name" onChange={(e) => store.setPassengerName(idx, e.target.value)}/>
                  <TextField value={p.surname} label="Surname" onChange={(e) => store.setPassengerSurname(idx, e.target.value)}/>
                  <TextField value={p.passport} label="Passport number" onChange={(e) => store.setPassengerPassport(idx, e.target.value)}/>
                  <PhoneNumberInput 
                    value={p.phoneNumber} 
                    defaultCountry={'ua'} 
                    variant="filled"
                    inputClass="phone-input"
                    onChange={(value) => {
                    if (typeof value === "string") {
                      store.setPassengerPhoneNumber(idx, value)
                    }
                  }} />
                </Box>
              </CardContent>
            </Card>
          ))}
          <Button
            variant='outlined'
            onClick={() => store.addPassenger()}
            sx={{
              width: "150px"
            }}
          >
            Add passenger
          </Button>
          <Button
            variant='contained'
            onClick={() => store.submit(authToken)}
            sx={{
              width: "300px",
              alignSelf: "center"
            }}
          >
            Reserve
          </Button>
        </>
      }
    </Box>
  )
});

export default NewBooking;