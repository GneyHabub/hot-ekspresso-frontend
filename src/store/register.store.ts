import { makeAutoObservable, runInAction } from "mobx";
import { register } from "../api/authentication";
import { FetchingStatus } from "../utils/types";
import { notificationStore } from "./notification.store";
import store from "./store";

class RegisterStore {
  fetchingStatus: FetchingStatus = "notFetched";
  name: string = '';
  nameDirty: boolean = false;
  email: string = '';
  emailDirty: boolean = false;
  password: string = '';
  passwordDirty: boolean = false;
  repeatPassword: string = '';

  constructor() {
    makeAutoObservable(this);
    this.submit = this.submit.bind(this);
  }

  setName(name: string) {
    this.name = name;
  }
  dirtyName() {
    this.nameDirty = true;
  }
  setEmail(email: string) {
    this.email = email;
  }
  dirtyEmail() {
    this.emailDirty = true;
  }
  setPassword(password: string) {
    this.password = password;
  }
  dirtyPassword() {
    this.passwordDirty = true;
  }
  setRepeatPassword(repeatPassword: string) {
    this.repeatPassword = repeatPassword;
  }
  async submit() {
    runInAction(() => {
      this.fetchingStatus = "fetching";
    });
    try {
      const token = await register({
        name: this.name,
        email: this.email, 
        password: this.password
      });
      store.setToken(token);
      runInAction(() => {
        this.fetchingStatus = "fetching";
      })
    } catch(e) {
      notificationStore.setNotificationFromError(e);
      this.fetchingStatus = "errorFetching";
    }
  }

  get isFetching() {
    return this.fetchingStatus === "fetching"
  }
  get passwordsMatch() {
    return this.password.localeCompare(this.repeatPassword) === 0;
  }
  get nameError() {
    return !this.name;
  }
  get emailError() {
    return !this.email;
  }
  get passwordError() {
    return !this.password;
  }
}

export default new RegisterStore();