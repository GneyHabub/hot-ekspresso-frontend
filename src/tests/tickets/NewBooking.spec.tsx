import { render } from '@testing-library/react';
import { BrowserRouter } from 'react-router-dom';
import { LocalizationProvider } from '@mui/lab';
import DateAdapter from '@mui/lab/AdapterDayjs';
import React from 'react';
import store from '../../store/newBookingForm.store';
import NewBooking from '../../pages/tickets/NewBooking';

jest.mock('../../store/newBookingForm.store', () => ({
  fetchAirports: jest.fn,
  resetForm: jest.fn,
  airportsfetchingStatus: 'fetched',
  airports: [
    {
      city: 'Moscow',
      name: 'Domodedovo',
      iata: 'DME',
    },
    {
      city: 'Moscow',
      name: 'Vnukovo',
      iata: 'VKO',
    },
    {
      city: 'Kazan',
      name: 'Kazan',
      iata: 'KZN',
    },
    {
      city: 'Saratov',
      name: 'Gagarin',
      iata: 'GSV',
    },
  ],
  setFromAirport: jest.fn,
  setToAirport: jest.fn,
  departureDate: new Date('1995-12-17T03:24:00'),
  setDepartureDate: jest.fn,
  serviceClass: 'Business',
  setServiceClass: jest.fn(),
  formIsValid: true,
  setSelectedFlight: jest.fn,
  searchFlights: jest.fn,
  selectedFlight: {
    id: 'AAA',
    price: 300,
    flights: [{
      from: 'KZN',
      to: 'GSV',
      departure: new Date('1995-12-17T03:24:00'),
      arrival: new Date('1995-12-17T03:24:00'),
      carrier: 'RedWings',
    }],
    serviceClass: 'Economy',
  },
  passengers: [{
    name: '',
    surname: '',
    passport: '',
    phoneNumber: '',
  }],
  flights: [
    {
      id: 'AAA',
      price: 300,
      flights: [{
        from: 'KZN',
        to: 'GSV',
        departure: new Date('1995-12-17T03:24:00'),
        arrival: new Date('1995-12-17T03:24:00'),
        carrier: 'RedWings',
      }],
      serviceClass: 'Economy',
    },
    {
      id: 'BBB',
      price: 700,
      flights: [
        {
          from: 'GSV',
          to: 'KZN',
          departure: new Date('1995-12-17T03:24:00'),
          arrival: new Date('1995-12-17T03:24:00'),
          carrier: 'RedWings',
        },
        {
          from: 'KZN',
          to: 'DME',
          departure: new Date('1995-12-17T03:24:00'),
          arrival: new Date('1995-12-17T03:24:00'),
          carrier: 'Pobeda',
        },
      ],
      serviceClass: 'Business',
    },
  ],
}));

describe('NewBooking', () => {
  it('matches the snapshot', () => {
    const wrapper = render(
      <LocalizationProvider dateAdapter={DateAdapter}>
        <BrowserRouter>
          <NewBooking />
        </BrowserRouter>
      </LocalizationProvider>,
    );
    expect(wrapper.baseElement).toMatchSnapshot();
  });
});
