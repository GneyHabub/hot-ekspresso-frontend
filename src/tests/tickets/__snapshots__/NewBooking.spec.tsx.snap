// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login matches the snapshot 1`] = `
<body>
  <div>
    <div
      class="MuiBox-root css-1q94rgb"
    >
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiTypography-root MuiTypography-h4 MuiTypography-gutterBottom css-1vw6mcs-MuiTypography-root"
          >
            New booking
          </div>
          <div
            class="MuiBox-root css-u6ojzc"
          >
            <div
              class="MuiFormControl-root form-block css-18simf8-MuiFormControl-root"
            >
              <div
                class="MuiBox-root css-0"
              >
                <div
                  aria-expanded="false"
                  class="MuiAutocomplete-root css-x1xt5h-MuiAutocomplete-root"
                  role="combobox"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="mui-2"
                      id="mui-2-label"
                    >
                      From
                      <span
                        aria-hidden="true"
                        class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-invalid="false"
                        autocapitalize="none"
                        autocomplete="off"
                        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                        id="mui-2"
                        required=""
                        spellcheck="false"
                        type="text"
                        value=""
                      />
                      <div
                        class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-173wfxe"
                        >
                          <span>
                            From
                             
                            *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
                <div
                  aria-expanded="false"
                  class="MuiAutocomplete-root css-x1xt5h-MuiAutocomplete-root"
                  role="combobox"
                >
                  <div
                    class="MuiFormControl-root MuiFormControl-fullWidth MuiTextField-root css-wb57ya-MuiFormControl-root-MuiTextField-root"
                  >
                    <label
                      class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary Mui-required css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                      data-shrink="false"
                      for="mui-4"
                      id="mui-4-label"
                    >
                      To
                      <span
                        aria-hidden="true"
                        class="MuiInputLabel-asterisk MuiFormLabel-asterisk css-wgai2y-MuiFormLabel-asterisk"
                      >
                         
                        *
                      </span>
                    </label>
                    <div
                      class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-fullWidth MuiInputBase-formControl MuiInputBase-adornedEnd MuiAutocomplete-inputRoot css-154xyx0-MuiInputBase-root-MuiOutlinedInput-root"
                    >
                      <input
                        aria-autocomplete="list"
                        aria-invalid="false"
                        autocapitalize="none"
                        autocomplete="off"
                        class="MuiOutlinedInput-input MuiInputBase-input MuiInputBase-inputAdornedEnd MuiAutocomplete-input MuiAutocomplete-inputFocused css-nxo287-MuiInputBase-input-MuiOutlinedInput-input"
                        id="mui-4"
                        required=""
                        spellcheck="false"
                        type="text"
                        value=""
                      />
                      <div
                        class="MuiAutocomplete-endAdornment css-1q60rmi-MuiAutocomplete-endAdornment"
                      />
                      <fieldset
                        aria-hidden="true"
                        class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                      >
                        <legend
                          class="css-173wfxe"
                        >
                          <span>
                            To
                             
                            *
                          </span>
                        </legend>
                      </fieldset>
                    </div>
                  </div>
                </div>
              </div>
              <div
                class="MuiBox-root css-0"
              >
                <div
                  class="MuiFormControl-root MuiTextField-root css-1g1iitx-MuiFormControl-root-MuiTextField-root"
                >
                  <label
                    class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-shrink MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary MuiFormLabel-filled css-1sumxir-MuiFormLabel-root-MuiInputLabel-root"
                    data-shrink="true"
                    for="mui-5"
                    id="mui-5-label"
                  >
                    Departure Date
                  </label>
                  <div
                    class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
                  >
                    <input
                      aria-invalid="false"
                      aria-label="Choose date, selected date is Dec 17, 1995"
                      aria-readonly="true"
                      class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                      id="mui-5"
                      readonly=""
                      type="text"
                      value="17/12/1995"
                    />
                    <fieldset
                      aria-hidden="true"
                      class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                    >
                      <legend
                        class="css-1z7n62"
                      >
                        <span>
                          Departure Date
                        </span>
                      </legend>
                    </fieldset>
                  </div>
                </div>
                <div
                  class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-1h5q5yu-MuiInputBase-root-MuiOutlinedInput-root"
                >
                  <div
                    aria-expanded="false"
                    aria-haspopup="listbox"
                    class="MuiSelect-select MuiSelect-outlined MuiOutlinedInput-input MuiInputBase-input css-11u53oe-MuiSelect-select-MuiInputBase-input-MuiOutlinedInput-input"
                    role="button"
                    tabindex="0"
                  >
                    Business
                  </div>
                  <input
                    aria-hidden="true"
                    class="MuiSelect-nativeInput css-yf8vq0-MuiSelect-nativeInput"
                    tabindex="-1"
                    value="Business"
                  />
                  <svg
                    aria-hidden="true"
                    class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium MuiSelect-icon MuiSelect-iconOutlined css-hfutr2-MuiSvgIcon-root-MuiSelect-icon"
                    data-testid="ArrowDropDownIcon"
                    focusable="false"
                    viewBox="0 0 24 24"
                  >
                    <path
                      d="M7 10l5 5 5-5z"
                    />
                  </svg>
                  <fieldset
                    aria-hidden="true"
                    class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                  >
                    <legend
                      class="css-nnbavb"
                    >
                      <span
                        class="notranslate"
                      >
                        ​
                      </span>
                    </legend>
                  </fieldset>
                </div>
              </div>
            </div>
            <button
              class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-sghohy-MuiButtonBase-root-MuiButton-root"
              tabindex="0"
              type="submit"
            >
              <span
                class="MuiButton-startIcon MuiButton-iconSizeMedium css-1d6wzja-MuiButton-startIcon"
              >
                <svg
                  aria-hidden="true"
                  class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
                  data-testid="SearchIcon"
                  focusable="false"
                  viewBox="0 0 24 24"
                >
                  <path
                    d="M15.5 14h-.79l-.28-.27C15.41 12.59 16 11.11 16 9.5 16 5.91 13.09 3 9.5 3S3 5.91 3 9.5 5.91 16 9.5 16c1.61 0 3.09-.59 4.23-1.57l.27.28v.79l5 4.99L20.49 19l-4.99-5zm-6 0C7.01 14 5 11.99 5 9.5S7.01 5 9.5 5 14 7.01 14 9.5 11.99 14 9.5 14z"
                  />
                </svg>
              </span>
              Search flights
              <span
                class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
              />
            </button>
          </div>
        </div>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation8 MuiCard-root css-t7h509-MuiPaper-root-MuiCard-root"
      >
        <div
          class="css-gmwslw-MuiStack-root"
        >
          <div
            class="css-6iefu9-MuiStack-root"
          >
            <h4
              class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
            >
              KZN
            </h4>
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeMedium css-i4bv87-MuiSvgIcon-root"
              data-testid="ArrowForwardIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="m12 4-1.41 1.41L16.17 11H4v2h12.17l-5.58 5.59L12 20l8-8z"
              />
            </svg>
            <h4
              class="MuiTypography-root MuiTypography-h4 css-5lbw0b-MuiTypography-root"
            >
              GSV
            </h4>
          </div>
          <button
            class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium css-1jfmqnk-MuiButtonBase-root-MuiIconButton-root"
            tabindex="0"
            type="button"
          >
            <svg
              aria-hidden="true"
              class="MuiSvgIcon-root MuiSvgIcon-fontSizeLarge css-tzssek-MuiSvgIcon-root"
              data-testid="CancelIcon"
              focusable="false"
              viewBox="0 0 24 24"
            >
              <path
                d="M12 2C6.47 2 2 6.47 2 12s4.47 10 10 10 10-4.47 10-10S17.53 2 12 2zm5 13.59L15.59 17 12 13.41 8.41 17 7 15.59 10.59 12 7 8.41 8.41 7 12 10.59 15.59 7 17 8.41 13.41 12 17 15.59z"
              />
            </svg>
            <span
              class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
            />
          </button>
        </div>
        <h6
          class="MuiTypography-root MuiTypography-subtitle1 css-10wpov9-MuiTypography-root"
        >
          17-12-1995
        </h6>
      </div>
      <div
        class="MuiPaper-root MuiPaper-elevation MuiPaper-rounded MuiPaper-elevation1 MuiCard-root css-bhp9pd-MuiPaper-root-MuiCard-root"
      >
        <div
          class="MuiCardContent-root css-46bh2p-MuiCardContent-root"
        >
          <div
            class="MuiBox-root css-gg4vpm"
          >
            <h6
              class="MuiTypography-root MuiTypography-h6 MuiTypography-gutterBottom css-18k87ye-MuiTypography-root"
            >
              Passenger #1
            </h6>
          </div>
          <div
            class="MuiBox-root css-16ejmj8"
          >
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-7"
                id="mui-7-label"
              >
                Name
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-7"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      Name
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-8"
                id="mui-8-label"
              >
                Surname
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-8"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      Surname
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <label
                class="MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined MuiFormLabel-root MuiFormLabel-colorPrimary css-14s5rfu-MuiFormLabel-root-MuiInputLabel-root"
                data-shrink="false"
                for="mui-9"
                id="mui-9-label"
              >
                Passport number
              </label>
              <div
                class="MuiOutlinedInput-root MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl css-9ddj71-MuiInputBase-root-MuiOutlinedInput-root"
              >
                <input
                  aria-invalid="false"
                  class="MuiOutlinedInput-input MuiInputBase-input css-1t8l2tu-MuiInputBase-input-MuiOutlinedInput-input"
                  id="mui-9"
                  type="text"
                  value=""
                />
                <fieldset
                  aria-hidden="true"
                  class="MuiOutlinedInput-notchedOutline css-1d3z3hw-MuiOutlinedInput-notchedOutline"
                >
                  <legend
                    class="css-173wfxe"
                  >
                    <span>
                      Passport number
                    </span>
                  </legend>
                </fieldset>
              </div>
            </div>
            <div
              class="MuiFormControl-root MuiTextField-root phone-input css-1u3bzj6-MuiFormControl-root-MuiTextField-root"
            >
              <div
                class="MuiFilledInput-root MuiFilledInput-underline MuiInputBase-root MuiInputBase-colorPrimary MuiInputBase-formControl MuiInputBase-adornedStart css-sgvi9r-MuiInputBase-root-MuiFilledInput-root"
              >
                <div
                  class="MuiInputAdornment-root MuiInputAdornment-positionStart MuiInputAdornment-filled MuiInputAdornment-sizeMedium MuiPhoneNumber-positionStart css-9mc7sf-MuiInputAdornment-root"
                >
                  <span
                    class="notranslate"
                  >
                    ​
                  </span>
                  <button
                    aria-haspopup="true"
                    aria-label="Select country"
                    class="MuiButtonBase-root MuiIconButton-root MuiIconButton-sizeMedium MuiPhoneNumber-flagButton css-78trlr-MuiButtonBase-root-MuiIconButton-root"
                    tabindex="0"
                    type="button"
                  >
                    <svg
                      class="margin"
                      viewBox="0 85.333 512 341.333"
                    >
                      <path
                        d="M0 85.337h512v341.326H0z"
                        fill="#FFDA44"
                      />
                      <path
                        d="M0 85.337h512V256H0z"
                        fill="#338AF3"
                      />
                    </svg>
                    <span
                      class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
                    />
                  </button>
                </div>
                <input
                  aria-invalid="false"
                  class="MuiFilledInput-input MuiInputBase-input MuiInputBase-inputAdornedStart css-1vbc0rj-MuiInputBase-input-MuiFilledInput-input"
                  id="mui-10"
                  placeholder="+1 (702) 123-4567"
                  type="tel"
                  value="+380"
                />
              </div>
            </div>
          </div>
        </div>
      </div>
      <button
        class="MuiButton-root MuiButton-outlined MuiButton-outlinedPrimary MuiButton-sizeMedium MuiButton-outlinedSizeMedium MuiButtonBase-root  css-e35vbc-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Add passenger
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
      <button
        class="MuiButton-root MuiButton-contained MuiButton-containedPrimary MuiButton-sizeMedium MuiButton-containedSizeMedium MuiButtonBase-root  css-1p94qe2-MuiButtonBase-root-MuiButton-root"
        tabindex="0"
        type="button"
      >
        Reserve
        <span
          class="MuiTouchRipple-root css-8je8zh-MuiTouchRipple-root"
        />
      </button>
    </div>
  </div>
</body>
`;
